src/OrchardCore.Modules/OrchardCore.Contents/Sitemaps/ContentTypesSitemapSourceBuilder.cs

```diff
@@ -14,51 +16,100 @@ public class ContentTypesSitemapSourceBuilder : SitemapSourceBuilderBase<Content
 
     private readonly IRouteableContentTypeCoordinator _routeableContentTypeCoordinator;
     private readonly IContentManager _contentManager;
+    private readonly YesSqlOptions _yesSqlOptions;
     private readonly IContentItemsQueryProvider _contentItemsQueryProvider;
     private readonly IEnumerable<ISitemapContentItemExtendedMetadataProvider> _sitemapContentItemExtendedMetadataProviders;
 
     public ContentTypesSitemapSourceBuilder(
         IRouteableContentTypeCoordinator routeableContentTypeCoordinator,
         IContentManager contentManager,
+        IOptions<YesSqlOptions> yesSqlOptions,
         IContentItemsQueryProvider contentItemsQueryProvider,
-        IEnumerable<ISitemapContentItemExtendedMetadataProvider> sitemapContentItemExtendedMetadataProviders
-        )
+        IEnumerable<ISitemapContentItemExtendedMetadataProvider> sitemapContentItemExtendedMetadataProviders)
     {
         _routeableContentTypeCoordinator = routeableContentTypeCoordinator;
         _contentManager = contentManager;
+        _yesSqlOptions = yesSqlOptions.Value;
         _contentItemsQueryProvider = contentItemsQueryProvider;
         _sitemapContentItemExtendedMetadataProviders = sitemapContentItemExtendedMetadataProviders;
     }
 
     public override async Task BuildSourceAsync(ContentTypesSitemapSource source, SitemapBuilderContext context)
     {
-        var queryContext = new ContentItemsQueryContext();
-        await _contentItemsQueryProvider.GetContentItemsAsync(source, queryContext);
-
         foreach (var sciemp in _sitemapContentItemExtendedMetadataProviders)
         {
             context.Response.ResponseElement.Add(sciemp.GetExtendedAttribute);
         }
 
-        foreach (var contentItem in queryContext.ContentItems)
+        var queryContext = new ContentItemsQueryContext()
+        {
+            Take = _yesSqlOptions.CommandsPageSize,
```

Comment:
Totally not the same thing, should not be used here.

Reply:
removed.